internal/buildengine/engine.go

```diff
@@ -786,6 +835,58 @@ func (e *Engine) getDependentModuleNames(moduleName string) []string {
 	return maps.Keys(dependentModuleNames)
 }
 
+// processDeployments runs in a separate goroutine and processes deployments from the queue
+func (e *Engine) processDeployments(ctx context.Context) {
+	logger := log.FromContext(ctx)
+	ticker := time.NewTicker(100 * time.Millisecond)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			if e.deploymentQueue.isEmpty() {
+				continue
+			}
+
+			queuedModules := e.deploymentQueue.takeAll()
+			logger.Debugf("Processing %d modules from deployment queue", len(queuedModules))
+
+			// Group modules by deployment settings
+			settingsGroups := make(map[string][]queuedModule)
+			for _, qm := range queuedModules {
+				key := fmt.Sprintf("%d-%v", qm.replicas, qm.waitForDeployOnline)
+				settingsGroups[key] = append(settingsGroups[key], qm)
+			}
+
+			// Process each group separately
+			for _, modules := range settingsGroups {
+				settings := modules[0] // Use first module's settings as they're all the same in this group
+				logger.Debugf("Deploying %d modules with replicas=%d, wait=%v", len(modules), settings.replicas, settings.waitForDeployOnline)
+
+				// Convert queuedModules to Modules for Deploy
+				moduleList := make([]Module, len(modules))
+				for i, qm := range modules {
+					moduleList[i] = qm.module
+				}
+
+				err := Deploy(ctx, e.projectConfig, moduleList, settings.replicas, settings.waitForDeployOnline, e.deployClient, e.schemaServiceClient)
+
+				// Signal completion to all modules in this group
+				for _, qm := range modules {
+					if err != nil {
```

Comment:
This if check can be skipped, as in both cases you are sending the value of `err`